src/connection.rs

```diff
@@ -103,18 +103,28 @@ impl Connection {
                                 .start_send(tokio_tungstenite::tungstenite::Message::text(json_msg))
                             {
                                 Ok(_) => {
-                                    assert!(matches!(
-                                        self.pending_requests.insert(msg.reference, sender),
-                                        None
-                                    ));
+                                    if self
+                                        .pending_requests
+                                        .insert(msg.reference, sender)
+                                        .is_some()
+                                    {
+                                        log::debug!("Replaced request with same ref");
+                                    }
                                 }
                                 Err(e) => {
-                                    let _ = sender.send(Err(RequestSendError::Send(e)));
+                                    if sender.send(Err(RequestSendError::Send(e))).is_err() {
+                                        log::debug!("Internal request channel receiver dropped");
+                                    }
                                 }
                             }
                         }
                         Err(e) => {
-                            let _ = sender.send(Err(RequestSendError::InvalidMessageOutJson(e)));
+                            if sender
+                                .send(Err(RequestSendError::InvalidMessageOutJson(e)))
+                                .is_err()
+                            {
+                                log::debug!("Internal request channel receiver dropped");
```

Comment:
When can this happen?

Reply:
If the request payload serialization failed